### DISCUSSION TITLE
ComfyUI  only using 50% of my VRAM?

### DISCUSSION THREAD

**Discussion by ZeroCool22** [Comfy-Org/ComfyUI#1043] [Q&A] [answered]
![Screenshot_2](https://github.com/comfyanonymous/ComfyUI/assets/13344308/8baaa4ae-036e-480b-bc8c-88e0da5cb339)

Come explanation for this?

With AUTO's it use about 9.5 of VRAM.

**Comment by ltdrdata** [comment]
Comfy follows a policy of moving data that is not being actively used by the GPU to RAM to maximize VRAM space. However, if your VRAM is sufficiently large, you can enable the --highvram option to use VRAM exclusively.

However, it is important to be mindful that as the workflow becomes longer and involves using various models, there is a possibility of encountering sudden VRAM OOM.

**Reply by ZeroCool22** [reply]
VRAM shortages, you mean OOM OUT OF RAM errors, but nothing that can break the card right?

**Reply by ZeroCool22** [reply]
And can you tell me where i add that argument?


I use the BAT to start the GUI.

**Reply by ltdrdata** [reply, marked as answer]
Yeah OOM. Nothing breaks card physically.
Just add --highvram in .bat file.

**Reply by ZeroCool22** [reply]
![Screenshot_4](https://github.com/comfyanonymous/ComfyUI/assets/13344308/007a5ff2-6cbf-43ec-919f-68f2126ab9c0)

Like this?

**Reply by ZeroCool22** [reply]
![Screenshot_5](https://github.com/comfyanonymous/ComfyUI/assets/13344308/e4d926e9-74b2-4841-bd71-bf88f8b26fd4)

Thx, a little better now, considering it's an old card.

**Reply by ZeroCool22** [reply]
Spoked too soon:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 144, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 74, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 67, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\nodes.py", line 241, in decode
    return (vae.decode(samples["samples"]), )
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\sd.py", line 615, in decode
    pixel_samples = self.decode_tiled_(samples_in)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\sd.py", line 582, in decode_tiled_
    (utils.tiled_scale(samples, decode_fn, tile_x // 2, tile_y * 2, overlap, upscale_amount = 8, pbar = pbar) +
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\utils.py", line 342, in tiled_scale
    ps = function(s_in).cpu()
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\sd.py", line 580, in <lambda>
    decode_fn = lambda a: (self.first_stage_model.decode(a.to(self.vae_dtype).to(self.device)) + 1.0).float()
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\ldm\models\autoencoder.py", line 94, in decode
    dec = self.decoder(z)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\model.py", line 727, in forward
    h = self.up[i_level].upsample(h)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\ComfyUI\comfy\ldm\modules\diffusionmodules\model.py", line 59, in forward
    x = torch.nn.functional.interpolate(x, scale_factor=2.0, mode="nearest")
  File "C:\Users\ZeroCool22\Desktop\ComfyUI_windows_portable\python_embeded\lib\site-packages\torch\nn\functional.py", line 3931, in interpolate
    return torch._C._nn.upsample_nearest2d(input, output_size, scale_factors)
torch.cuda.OutOfMemoryError: Allocation on device 0 would exceed allowed memory. (out of memory)
Currently allocated     : 9.98 GiB
Requested               : 224.00 MiB
Device limit            : 11.00 GiB
Free (according to CUDA): 0 bytes
PyTorch limit (set by user-supplied memory fraction)
                        : 17179869184.00 GiB
```


So, i'm stuck at using only 50% of the VRAM or nothing.

**Comment by tjedwards** [comment]
I've got 16GB of VRAM (on a 2080), and normal mode uses half of it but only while working on an image.  I tried --gpu-only (OOM results) and --highvram.  The latter works, but is actually over 20x slower (490sec vs 20sec)!  In normal mode my system RAM is high and my SSD is busy swapping.  I would have expected that to be slower, but it's not; very strange.

**Comment by 42degrees** [comment]
I'm seeing similar results, I have 24GB VRAM (RTX 3090) and 32GB of RAM and with --highvram made no difference in VRAM usage.  

I'm generating batch of one, 4096X2048.  Sometimes I get a result, sometimes just polka dots (maybe that's what I get for trying to do this large an image on non SDXL checkpoints?).  This render takes 4-5 minutes for SD 1.5 or SDXL 1.0, but SD 1.4 took twice as long.

Using the v1.5 pruned EMA Only checkpoint.  Euler with 60 steps.  It never went above 8.4GB of VRAM usage.
Using the SDXL base 1.0 checkpoint with the same settings uses 11.9GB of VRAM.
Using SD V1.4 Full EMA checkpoint with the same settings uses 8.5GB of VRAM.

It's using under a GB of RAM (total usage is 21GB out of 32GB RAM).  I see only a 15-20% CPU usage, 1-9% GPU usage.

I get, "Warning: Ran out of memory when regular VAE decoding, retrying with tiled VAE decoding." for all renders.  But, of course, I'm nowhere near out of memory or video memory.

If I lower the image to 2048x1024 it only takes 39 seconds to render and I get no out of memory error from the VAE decoder.  With SDXL base 1.0 it never used more than 8GB of memory.

Note that I'm new to SD and so I don't know what settings would actually impact VRAM usage.

If you care this is the prompt I tested with:
Color panoramic landscape wide angle photography, disaster movie, entire valley, (space elevator:2.0), one erupting volcano throwing large burning rocks in distance, cloud lightning, Chicago, damaged and burning buildings, lakeside, sunrise, dramatic lighting, puffy red and purple clouds, god rays, (2000mm lens f2.5).

Note that ComfyUI is detecting my correct configuration:
P:\ComfyUI>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build --highvram
Total VRAM 24576 MB, total RAM 32517 MB
xformers version: 0.0.21
Set vram state to: HIGH_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
VAE dtype: torch.bfloat16

**Comment by C-Fu** [comment]
Just to add, using SD1.5 fixed it for me. xl threw errors, 2.1 threw errors.

**Comment by cmcm0046** [comment]
执行VRAM_Debug时出错：

VRAM_Debug.VRAMdebug（） 有一个意外的关键字参数“image_passthrough”

文件“I：\comfyui\execution.py”，第 151 行，recursive_execute
output_data，output_ui = get_output_data（obj， input_data_all）
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
文件“I：\comfyui\execution.py”，第 81 行，get_output_data
return_values = map_node_over_list（obj， input_data_all， obj.函数，allow_interrupt=True）
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
文件“I：\comfyui\execution.py”，第 74 行，map_node_over_list
results.append（getattr（obj， func）（**slice_dict（input_data_all， i）））

**Reply by cmcm0046** [reply]
i can't understand, what kind of mistake is it?
